netaddr_test.go

```diff
@@ -363,10 +363,16 @@ func TestIPMarshalUnmarshalBinary(t *testing.T) {
 	}
 
 	// Cannot unmarshal into a non-zero IP
-	ip := MustParseIP("1.2.3.4")
-	if err := ip.UnmarshalBinary([]byte{1, 1, 1, 1}); err == nil {
+	ip1 := MustParseIP("1.2.3.4")
+	if err := ip1.UnmarshalBinary([]byte{1, 1, 1, 1}); err == nil {
 		t.Fatal("unmarshaled into non-empty IP")
 	}
+
+	// Cannot unmarshal from unexpected IP length.
```

Comment:
Should there be  a test for unmarshaling from too long of a buffer?

Reply:
Sure, done!

Reply:
Oh sorry, did you mean >16 bytes? If so that's covered by the IPv6+zone cases above. But will see if you were thinking the same thing I was before reverting the loop.

Reply:
3B and 5B were the cases I was thinking of.

Reply:
Nice, any objections to the updated code?

Reply:
None from me.